subfun/BIDS_FFX.m

```diff
@@ -94,67 +94,67 @@ function BIDS_FFX(action, degreeOfSmoothing, opt)
                 matlabbatch{1}.spm.stats.fmri_spec.dir = {ffxDir};
                 % FMRI DESIGN
                 fprintf(1,'BUILDING JOB : FMRI design\n')
-                
+
                 sesCounter = 1;
-                
-                
+
+
                 %%
                 % identify sessions for this subject
                 [sessions, numSessions] = getSessions(BIDS, subNumber, opt);
-                
+
                 for iSes = 1:numSessions % For each session
-                    
+
                     % get all runs for that subject across all sessions
                     [runs, numRuns] = getRuns(BIDS, subNumber, sessions{iSes}, opt);
-                    
+
                     for iRun = 1:numRuns % For each run
-                        
+
                         fprintf(1,'PROCESSING GROUP: %s SUBJECT No.: %i SUBJECT ID : %s SESSION: %i RUN:  %i \n',...
                             groupName,iSub,subNumber,iSes,iRun)
-                        
+
                         % get the filename for this bold run for this task
                         [fileName, subFuncDataDir]= getBoldFilename(...
                             BIDS, ...
                             subNumber, sessions{iSes}, runs{iRun}, opt);
-                        
+
                         % check that the file with the right prefix exist
                         files = inputFileValidation(subFuncDataDir, prefix, fileName);
-                        
+
                         disp(files)
-                        
+
                         tsvFile{sesCounter,1} = fullfile(subFuncDataDir, ...
                             [fileName(1:end-9),'_events.tsv']); %#ok<*AGROW>
                         rpFile{sesCounter,1} = ...
                             fullfile(subFuncDataDir, ...
                             ['rp_', motionRegressorPrefix, fileName(1:end-4),'.txt']);
-                        
-                        
-                        
-                        
-                        
-                        
-                        
-                        
-                        
-                        
+
+
+
+
+
+
+
+
+
+
                         % Convert the tsv files to a mat file to be used by SPM
                         % DO WE NEED TO TRIM ONSET BY A DURATION EQUIVALENT
                         % TO THE NUMBER OF DUMMIES REMOVED
                         convertTsv2mat(tsvFile{sesCounter,1})
```

Comment:
No, We dont need to do that, the Experiment start time begins after the dummy scans are over (at least thats how we design the stimuli delivery scripts/experiments). So the time doesnt need to be corrected. Its already ok.

Reply:
it also depends how you prepare you experiment/stimulation scripts. 
For example in all our previous experiments, the script gets the initial dummy scans THEN the timing of the experiment starts . So Time ZERO is already after the dummies are removed. At least this is the current siutation, and if this is going to change, we know to agree within the lab that all the stimulation scripts should change accordingly in the future.

Check it and let me know. If there is no specs for that and its up to us, I would vote to keep it like now. Its easier and less editing with the onsets. Lets see.

Reply:
Here is what the specs say:

`Onset (in seconds) of the event measured from the beginning of the acquisition of the first volume in the corresponding task imaging data file. If any acquired scans have been discarded before forming the imaging data file, ensure that a time of 0 corresponds to the first image stored. In other words negative numbers in "onset" are allowed.`

https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/05-task-events.html

So everything is relative to the first image in the BIDS data set. So either the onset times have to be offset by the time equivalent to the number of dummies to be discarded or we have to remove the dummies from the BIDS data set itself.

I think the second one is safer.

Does that make sense?

Reply:
it makes sense, but it is also speaking about the data that will be in the BIDS format (The 4D nifti). so if we start the timing from the 1st image including the dummies, the we should never run the remove dummies. 
If we remove the dummies, we are deleting the first 4 volumes but the onsets will be off by 4 volumes. 
 
They are also saying " ensure that a time of 0 corresponds to the first image stored. In other words negative numbers in "onset" are allowed." 
negative numbers are allowed ---> probably reffering to dummies. 
we can discuss this point in detail in the meeting. 

For the workshop --> the BIDS folder can have the 4D with the removed dummies, and this way the onsets will be ok without a problem.

Reply:
> They are also saying " ensure that a time of 0 corresponds to the first image stored. In other words negative numbers in "onset" are allowed."
> negative numbers are allowed ---> probably reffering to dummies.
> we can discuss this point in detail in the meeting. 

There is a footnote in the specs and it seems that negative onsets were more there for training trials that might happen before the "real" things starts.

But yeah let's talk about this during the meeting.

> For the workshop --> the BIDS folder can have the 4D with the removed dummies, and this way the onsets will be ok without a problem.

Yup

Reply:
I will resolve this converstation for now.